### PR TITLE
WIP: Enhance the "cleanupdb" command

### PR DESCRIPTION
**Pull request by rjarry**
Following the discussion on #4536.

This is a bit tedious because of SQLite and MySQL with MyISAM... However, the idea is only to have one script to maintain that works "everywhere".

This is only a draft, open for discussion. I'm not sure how to test this properly.

### CODE REVIEW

File: master/buildbot/scripts/cleanupdb.py [hunk 1]
```diff
@@ -30,14 +33,127 @@
...
 
-    master = BuildMaster(config['basedir'])
-    master.config = master_cfg
-    db = master.db
-    yield db.setup(check_version=False, verbose=not config['quiet'])
+    def thd(conn):
+        with conn.begin():
+            # Delete sourcestamps older than --max-days
+            # The starting point is not "now" but the most recent sourcestamp..
+            max_created_at = conn.execute(
```
Comment (tardyp) [inline master/buildbot/scripts/cleanupdb.py]: so why not sql, like we planned?

File: master/buildbot/scripts/cleanupdb.py [hunk 2]
```diff
@@ -51,6 +167,31 @@ def doCleanupDatabase(config, master_cfg):
...
+    master = BuildMaster(config['basedir'])
+
+    master.config = master_cfg
+    db = master.db
+    yield db.setup(check_version=False, verbose=not config['quiet'])
+
+    for job, func in MAINTENANCE_JOBS:
+        if job in config['jobs']:
+            print('running job %s...' % job)
+            yield func(db, config)
```
Comment (tardyp) [inline master/buildbot/scripts/cleanupdb.py]: looks fine. This can be easily ported to janitor.

File: master/buildbot/scripts/cleanupdb.py (hunk 1 above)
Comment (tardyp) [reply master/buildbot/scripts/cleanupdb.py]: :+1:

**Comment by rjarry** [comment]
By the way, I did not give up on this. I'll try to work on it in the next few weeks.

File: master/buildbot/scripts/base.py [hunk 3]
```diff
@@ -277,6 +278,78 @@ def postOptions(self):
                 msg = 'Required argument missing: ' + missing[0]
             raise usage.UsageError(msg)
 
+    def getUsage(self, width=None):
```
Comment (p12tic) [inline master/buildbot/scripts/base.py]: I wonder, wouldn't it be possible to use the standard `argparse` module? It has support for subcommands.

**Pull request closed without merging**